### PR TITLE
Update Windows docs

### PR DESCRIPTION
**Pull request by lmm**
## Description

This PR fixes some issues in the Calico for Windows pages. Thanks to @lwr20 for some of these suggestions 🙂 

- Improved the EKS quickstart
- Platform support is updated to match what we test
- Supported Windows Server version updated to match across the various pages
- Misc spelling/wording fixes

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: getting-started/windows-calico/kubernetes/requirements.md [hunk 1]
```diff
@@ -55,19 +58,17 @@ calicoctl patch felixconfiguration default -p '{"spec":{"ipipEnabled":false}}'
 calicoctl ipam configure --strictaffinity=true
 ```
 
->**Note**: {{site.prodnameWindows}} requires four Linux worker nodes in order to meet high-availability requirements for Typha.
```
Comment (lmm) [inline getting-started/windows-calico/kubernetes/requirements.md:58]: This was fixed a while ago in the operator

File: getting-started/windows-calico/kubernetes/requirements.md [hunk 2]
```diff
@@ -45,8 +48,8 @@ Earlier versions may work, but we do not actively test {{site.prodnameWindows}}
 
 #### Linux platform 
 
-- At least one Linux Kubernetes worker node to run {{site.prodname}}'s cluster-wide components that meets [Linux system requirements]({{site.baseurl}}/getting-started/kubernetes/requirements), and is installed with {{site.prodname}} v3.12.0+.
-- VXLAN or BGP without encapsulation is supported if using Calico CNI. IPIP (default encapsulation mode) is not supported. Use the following command to turn off IPIP.
+- At least one Linux Kubernetes worker node to run {{site.prodname}}'s cluster-wide components that meets [Linux system requirements]({{site.baseurl}}/getting-started/kubernetes/requirements), and is installed with {{site.prodname}} v3.12+.
+- VXLAN or BGP without encapsulation is supported if using Calico CNI. IPIP (Calico's default encapsulation mode) is not supported. Use the following command to turn off IPIP.
```
Comment (song-jiang) [inline getting-started/windows-calico/kubernetes/requirements.md:52]: Shall we use {{site.prodname}} instead of Calico?

**Comment by song-jiang** [comment]
Just one comment and leave the PR to @lwr20 for approval.

File: getting-started/windows-calico/kubernetes/requirements.md (hunk 2 above)
Comment (lwr20) [reply getting-started/windows-calico/kubernetes/requirements.md:52]:
```suggestion
- VXLAN or BGP without encapsulation is supported if using {{site.prodname}} CNI. IPIP ({{site.prodname}}'s default encapsulation mode) is not supported. Use the following command to turn off IPIP.
```

**Review by lwr20 — APPROVED** [review]

**Pull request merged**